x-pack/plugins/security_solution/public/detection_engine/rule_management/api/api.ts

```diff
@@ -380,6 +412,35 @@ export async function performBulkAction({
   });
 }
 
+/**
+ * Perform bulk schedule backfill action with rules selected by a filter query
+ *
+ * @param dryRun enables dry run mode for bulk actions
+ *
+ * @throws An error if response is not OK
+ */
+export async function performBulkScheduleBackfillAction({
```

Comment:
The `performBulkAction` method runs bulk actions which modify the rule (create, update, delete etc.) and returns the corresponding results object:

```
export interface BulkActionResult {
  updated: RuleResponse[];
  created: RuleResponse[];
  deleted: RuleResponse[];
  skipped: RuleResponse[];
}
```

At first, I tried to add a `backfilled` property to `BulkActionResult` and make `performBulkAction` a generic method which will cover backfill bulk actions, but then I decided not to go with this approach because:
1. I did not like the mixture of different unrelated actions. Right now `performBulkAction` handles everything in a way that it expects the `CRUD` actions. It reflected in types names and objects that we return.
2. When I tried to add `backfilled: RuleResponse[];` to the `BulkActionResult`, I've got an error about complexity of the type: `"The inferred type of this node exceeds the maximum length the compiler will serialize. An explicit type annotation is needed."`. To overcome it, we would need to re-structure  the whole `BulkActionResult` in a way to make it simpler for the typescript compiler and to make sure that we do not break current APIs.

Reply:
Did you consider here reusing "updated" instead of creating a new `backfilled` prop? Do you think it's just wrong from an API design point of view? (I understand your point about CRUD vs scheduling a run)

My point here is that reusing `performBulkAction` also means not having to create a separate `buildBulkScheduleBackfillResponse` or modify the types, or create new mutations at all. If the cost to pay for that is only having an API response that is not as specific as it should be, then maybe it's a cost we want to pay.

Reply:
Updated to use `performBulkAction` for "manual rule run" action as well.